aws(eks): add EKS + Kubernetes Hello World

metral opened the pull request.

lukehoban commented on aws-ts-eks-hello-world/README.md:
````diff
@@ -0,0 +1,148 @@
+    ```
+
+1. After 10-15 minutes, your cluster will be ready, and the kubeconfig JSON you'll use to connect to the cluster will
+   be available as an output. You can save this kubeconfig to a file like so:
````
There seems to be a gap between the `like so: ` and where this is actually explained.  

Also don't quite follow the paragraph that follows this in context.

lblackstone replied:
```suggestion
   be available as an output.
```

lukehoban commented on aws-ts-eks-hello-world/README.md:
```diff
@@ -0,0 +1,148 @@
+    resource can be made in-place, and which require replacement, and computes
+    the minimally disruptive change to achieve the desired state.
+    
+    **Note:** Pulumi auto-generates a suffix for all objects. Pulumi's object model does
```
Do you think we have to reference this here?    In particular, how does this relate to the rest of the walkthrough?

lblackstone replied:
Yeah, I think it would be better to move the lengthier description to https://pulumi.io/reference/programming-model.html and just keep the first sentence with a link to the lengthier description.
```suggestion
    **Note:** By default, Pulumi [auto-generates a suffix for all objects](https://pulumi.io/reference/programming-model.html).
```

lukehoban commented on aws-ts-eks-hello-world/package.json:
```diff
@@ -0,0 +1,10 @@
+        "@types/node": "latest",
+        "@pulumi/aws": "latest",
+        "@pulumi/aws-infra": "^0.16.2",
+        "@pulumi/eks": "^0.16.6",
```
I suppose we may as well just make these all `latest`?

lukehoban commented on aws-ts-eks-hello-world/Pulumi.yaml:
```diff
@@ -0,0 +1,3 @@
+name: eks-hello-world
+runtime: nodejs
+description: A minimal AWS TypeScript Pulumi program
```
Probably should update this `description`.

lblackstone commented on aws-ts-eks-hello-world/README.md:
```diff
@@ -0,0 +1,148 @@
+    resource can be made in-place, and which require replacement, and computes
+    the minimally disruptive change to achieve the desired state.
+    
+    **Note:** Pulumi auto-generates a suffix for all objects. Pulumi's object model does
```
```suggestion
    **Note:** By default, Pulumi auto-generates a suffix for all objects. Pulumi's object model does
```

lblackstone commented on aws-ts-eks-hello-world/README.md:
```diff
@@ -0,0 +1,148 @@
+    guaranteed to have a differing, non-conflicting name. Doing this
+    allows a new resource to be created, and dependencies to be updated to
+    point to the new resource, before the old resource is deleted.
+    This is generally quite useful.
```
Pulumi will not auto-generate the name if `metadata.namespace` is set.

lblackstone commented on aws-ts-eks-hello-world/README.md:
```diff
@@ -0,0 +1,148 @@
+    $ pulumi stack output kubeconfig > kubeconfig
+    $ export KUBERNETES_VERSION=1.11.5 && sudo curl -s -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/amd64/kubectl && sudo chmod +x /usr/local/bin/kubectl
+
+    $ export KUBECONFIG=`pwd`/kubeconfig
```
```suggestion
    $ export KUBECONFIG=$PWD/kubeconfig
```

metral commented:
> LGTM
> 
> For bonus points, we should add this to `misc/test/examples_test.go` so that we get nightly test coverage of this example against head of all repos to provide (a) regression prevention for the libraries generally (b) validation that this specific example doesn't break in the future. Even something that doesn't (yet) use `ExtraRuntimeValidation` would be really valuable (and hopefully easy?).

I'll go ahead and tackle adding this to the integration tests in a separate PR to get this PR merged for https://github.com/pulumi/docs/pull/838

metral commented:
@lblackstone @lukehoban I've gone ahead and added in a k8s LB Service to the demo to:
- publicly expose the nginx deployment for a more richer experience OOTB for the user
- prime this for the integration tests in `misc/test/examples_test.go` as I can then issue a HTTP req to see if we 200 or not for success in a future PR

I also added in an interactive blurb for the user to pull in existing guestbook YAML manifest to provide a more in-depth experience that they can do such actions when considering using pulumi.

PTAL

lblackstone commented on aws-ts-eks-hello-world/README.md:
```diff
@@ -0,0 +1,139 @@
+   This is an example on how to create the standard Guestbook manifests in
+   Pulumi from its YAML manifests. We take the additional steps of transforming
+   its properties to use the same Namespace and metadata labels that
+   the NGINX stack uses, as well as, make its frontend service use a
```
```suggestion
   the NGINX stack uses, and also make its frontend service use a
```

metral commented on aws-ts-eks-hello-world/index.ts:
```diff
@@ -60,3 +60,25 @@ export const deployment = new k8s.apps.v1.Deployment(name,
 export const deploymentName = deployment.metadata.apply(m => m.name);
+
+// Create a LoadBalancer Service for the NGINX Deployment
+export const service = new k8s.core.v1.Service(name,
```
I was doing this for users who cared to see the actual k8s spec, thinking someone new to pulumi from k8s would find it informative, instead of say just seeing a generated resource name. I could certainly remove it if it feels extraneous.

metral replied:
I've gone ahead and removed the k8s resources from being exported.

metral commented:
Squashed commits
